### PR TITLE
Updated communication with backend Java program.

### PR DESCRIPTION
**Pull request by amgrubb**
- Create new conversion from graph object to Gson.
- Update ModelSpecBuilder with resulting objects.

### CODE REVIEW

File: leaf-analysis/src/gson_classes/IGraph.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package gson_classes;
+
+import java.util.List;
+
+
```
Comment (YesugenB) [inline leaf-analysis/src/gson_classes/IGraph.java:5]: Extra line here?

File: leaf-analysis/src/interface_objects/FuncWrapper.java [hunk 2]
```diff
@@ -11,4 +11,5 @@
  * This also makes converting the JSON to Java objects easier.
  */
 public interface FuncWrapper {
+
```
Comment (YesugenB) [inline leaf-analysis/src/interface_objects/FuncWrapper.java]: Will this be updated?

File: leaf-analysis/src/interface_objects/InputObject.java [hunk 3]
```diff
@@ -17,16 +17,16 @@ public InputAnalysis getAnalysis() {
 		return analysisRequest;
 	}
 	
-	public void setAnalysis(InputAnalysis analysis) {
-		this.analysisRequest = analysis;
-	}
+//	public void setAnalysis(InputAnalysis analysis) {
+//		this.analysisRequest = analysis;
+//	}
```
Comment (YesugenB) [inline leaf-analysis/src/interface_objects/InputObject.java:22]: Can these commented lines be taken out?

File: leaf-analysis/src/simulation/Actor.java [hunk 4]
```diff
@@ -15,6 +15,11 @@ public Actor(String nodeID, String nodeName, String nodeType){
 		actorType = nodeType;
 		// nodeType indicates whether it is an Actor, Role, or Agent.
 	}
+	public Actor(String nodeID, String nodeName, String nodeType, String uniqueID){
```
Comment (YesugenB) [inline leaf-analysis/src/simulation/Actor.java:18]: Are there supposed to be two `public Actor` here? Or is this method overloading 🧐

File: leaf-analysis/src/simulation/BIModelSpecBuilder.java [hunk 5]
```diff
@@ -0,0 +1,498 @@
+package simulation;
+
```
Comment (YesugenB) [inline leaf-analysis/src/simulation/BIModelSpecBuilder.java:2]: I think there are a lot of extra lines and odd indents in this file. Not sure if you wanted to change it later/have me point them all out.

File: leaf-ui/js/displayAnalysis.js [hunk 6]
```diff
@@ -126,41 +140,38 @@ function adjustSliderWidth(maxValue){
  * @param {Number} sliderValue
  *   Current value of the slider
  * @param {Number} currentValueLimit
- * @param {AnalysisResult} currentAnalysis
- *  an AnalysisResult object that contains data about the analysis that the back end performed
+ * @param {ResultBBM} currentAnalysis
+ *  an ResultBBM object that contains data about the analysis that the back end performed
```
Comment (YesugenB) [inline leaf-ui/js/displayAnalysis.js]: Should be `a ResultBBM object`

**Review by YesugenB — COMMENTED** [review]
I think this looks good from what I could tell? Just made some small comments.

**Review by YesugenB — APPROVED** [review]

File: leaf-analysis/src/gson_classes/IGraph.java (hunk 1 above)
Comment (amgrubb) [reply leaf-analysis/src/gson_classes/IGraph.java:5]: Fixed

File: leaf-analysis/src/interface_objects/FuncWrapper.java (hunk 2 above)
Comment (amgrubb) [reply leaf-analysis/src/interface_objects/FuncWrapper.java]: This is required to be as it is to make GSon work. No updates required.

File: leaf-analysis/src/interface_objects/InputObject.java (hunk 3 above)
Comment (amgrubb) [reply leaf-analysis/src/interface_objects/InputObject.java:22]: This file will eventually be deleted.

File: leaf-analysis/src/simulation/Actor.java (hunk 4 above)
Comment (amgrubb) [reply leaf-analysis/src/simulation/Actor.java:18]: Yes, this enables backwards compatibility and us to pass additional parameters.

File: leaf-analysis/src/simulation/BIModelSpecBuilder.java (hunk 5 above)
Comment (amgrubb) [reply leaf-analysis/src/simulation/BIModelSpecBuilder.java:2]: Yes, this file is only 1/4 done. I'll clean it up for the next PR. :) 
Thanks!!!

**Pull request merged**